src/statsig/internal/statsig_driver.rs

```diff
@@ -29,31 +31,33 @@ pub struct StatsigDriver {
 
 impl StatsigDriver {
     pub fn new(secret_key: &str, options: StatsigOptions) -> std::io::Result<Self> {
-        let runtime = match Builder::new_multi_thread()
-            .worker_threads(3)
-            .thread_name("statsig")
-            .enable_all()
-            .build()
-        {
-            Ok(rt) => rt,
-            Err(e) => {
-                return Err(e);
-            }
+        let (opt_runtime, handle) = if let Ok(handle) = Handle::try_current() {
+            (None, handle)
+        } else {
+            let rt = match Builder::new_multi_thread()
+                .worker_threads(3)
+                .thread_name("statsig")
+                .enable_all()
+                .build()
```

Comment:
Nit: can't you use the question mark here?

```
let rt = match Builder::new_multi_thread()
                .worker_threads(3)
                .thread_name("statsig")
                .enable_all()
                .build()?;
```

Reply:
Yep! nice catch